### PR TITLE
NETOBSERV-1259: Adding Netobserv DNS tracking blog

### PR DESCRIPTION
**Pull request by msherif1234**
Add `DNS tracking` blog

### CODE REVIEW

**Comment by jpinsonneau** [comment]
Added the UI parts :smile_cat: 
Your turn @memodi !

File: blogs/dns_tracking/dns_tracking.md [hunk 1]
```diff
@@ -61,13 +59,67 @@ collector object with the following fields enabled in eBPF config
...
+apiVersion: flows.netobserv.io/v1beta1
+kind: FlowCollector
+metadata:
+  name: cluster
+spec:
   agent:
     type: EBPF
     ebpf:
       privileged: true
-      enableDNSTracking: true
```
Comment (jpinsonneau) [inline blogs/dns_tracking/dns_tracking.md]: That should end like this:
https://github.com/netobserv/network-observability-operator/pull/394/files#r1304927133

Let's keep an eye on the final output

File: blogs/dns_tracking/dns_tracking.md [hunk 2]
```diff
@@ -61,13 +59,67 @@ collector object with the following fields enabled in eBPF config
...
+      features:
+        - DnsTacking
```
Comment (jpinsonneau) [inline blogs/dns_tracking/dns_tracking.md:72]: That should end like this:
https://github.com/netobserv/network-observability-operator/pull/394/files#r1304927133

Let's keep an eye on the final output

File: blogs/dns_tracking/dns_tracking.md [hunk 3]
```diff
@@ -61,13 +59,67 @@ collector object with the following fields enabled in eBPF config
...
+
+The first one will allow you to filter on a specific DNS Id to correlate with your query.
+
+![dns id](./images/dns_id.png)
+
+The second one helps to identify potential performance issues by looking at DNS resolution latency.
+
+![dns latency more than](./images/dns_latency_more_than.png)
+
+The third filter surfaces DNS response codes, which can help detect errors or unauthorized resolutions.
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:96]: what does "unauthorized resolutions" mean? or should we simplify say "which can help detect DNS error sources"

File: blogs/dns_tracking/dns_tracking.md [hunk 4]
```diff
@@ -61,13 +59,67 @@ collector object with the following fields enabled in eBPF config
...
+
+![advanced options 2](./images/advanced_options2.png)
+
+The DNS related flows will show these informations in both table and side panel:
+
+![dns table](./images/dns_table.png)
+
 ## Future support
 
 - looking at adding mDNS
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:125]: if this stems from the observations during testing, I don't think we saw anything about mDNS in our tests, eventually I realized the port 5353 we were seeing is the internal port that dns pods are using, the service is exposed via port 53

File: blogs/dns_tracking/dns_tracking.md [hunk 5]
```diff
@@ -61,13 +59,67 @@ collector object with the following fields enabled in eBPF config
...
+- Top 5 average DNS latencies
+- Top 5 DNS response code
+- Top 5 DNS response code stacked with total
+
+![dns graphs 1](./images/dns_graphs1.png)
+![dns graphs 2](./images/dns_graphs2.png)
+
+
+### Traffic flows
+The table view will get the new DNS related columns `Id`, `Latency` and `Response code` available from
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:114]: probably just "DNS columns" (-related-)

File: blogs/dns_tracking/dns_tracking.md [hunk 6]
```diff
@@ -61,13 +59,67 @@ collector object with the following fields enabled in eBPF config
...
+![dns graphs 2](./images/dns_graphs2.png)
+
+
+### Traffic flows
+The table view will get the new DNS related columns `Id`, `Latency` and `Response code` available from 
+the `advanced options` -> `manage columns` popup
+
+![advanced options 2](./images/advanced_options2.png)
+
+The DNS related flows will show these informations in both table and side panel:
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:119]: here. "The DNS flows"

**Review by memodi — CHANGES_REQUESTED** [review]

File: blogs/dns_tracking/dns_tracking.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+## Understanding DNS Resolution
+
+Before diving into the specifics of eBPF tracepoint hooks, let's briefly
+recap how DNS resolution works.
+When you enter a website's domain name (e.g., www.example.com) in your
+browser, your computer needs to find the corresponding IP address.
+This process involves multiple steps, including querying DNS servers,
+caching responses, and ultimately obtaining the IP address for
+establishing a connection.
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:26]:
```suggestion
This process involves multiple steps, including querying DNS servers, obtaining the IP address to establish a connection and caching response for future re-occurrence of same DNS query
```

File: blogs/dns_tracking/dns_tracking.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+## Utilizing Tracepoint Hooks for DNS Tracking
+
+Tracepoint hooks are predefined points in the Linux kernel where eBPF
+programs can be attached to capture and analyze specific events.
+For DNS tracking, we leveraged tracepoint hooks associated with DNS
+resolution processes specifically `tracepoint/net/net_dev_queue` tracepoint,
+Then we parse the DNS header to determine if its query or response, attempt
+to correlate query and response for specific DNS transaction and then record the
+elapsed time as well as enrich the flow with DNS header's information like DNS Id
+and DNS flags to help UI filtering.
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:37]:
```suggestion
and DNS response codes to help filtering in Network Observability console.
```

File: blogs/dns_tracking/dns_tracking.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+## Utilizing Tracepoint Hooks for DNS Tracking
+
+Tracepoint hooks are predefined points in the Linux kernel where eBPF
+programs can be attached to capture and analyze specific events.
+For DNS tracking, we leveraged tracepoint hooks associated with DNS
+resolution processes specifically `tracepoint/net/net_dev_queue` tracepoint,
+Then we parse the DNS header to determine if its query or response, attempt
+to correlate query and response for specific DNS transaction and then record the
+elapsed time as well as enrich the flow with DNS header's information like DNS Id
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:36]:
```suggestion
elapsed time to determine DNS latency as well as to enrich the flow with DNS header's information like DNS Id
```

File: blogs/dns_tracking/dns_tracking.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+![dns graphs 2](./images/dns_graphs2.png)
+
+
+### Traffic flows
+The table view will get the new DNS related columns `Id`, `Latency` and `Response code` available from 
+the `advanced options` -> `manage columns` popup
+
+![advanced options 2](./images/advanced_options2.png)
+
+The DNS related flows will show these informations in both table and side panel:
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:119]:
```suggestion
The DNS flows will show these informations in both table and side panel:
```

File: blogs/dns_tracking/dns_tracking.md [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+- Top 5 average DNS latencies
+- Top 5 DNS response code
+- Top 5 DNS response code stacked with total
+
+![dns graphs 1](./images/dns_graphs1.png)
+![dns graphs 2](./images/dns_graphs2.png)
+
+
+### Traffic flows
+The table view will get the new DNS related columns `Id`, `Latency` and `Response code` available from
```
Comment (memodi) [inline blogs/dns_tracking/dns_tracking.md:114]:
```suggestion
The table view will get the new DNS columns `Id`, `Latency` and `Response code` available from 
```

File: blogs/dns_tracking/dns_tracking.md (hunk 3 above)
Comment (msherif1234) [reply blogs/dns_tracking/dns_tracking.md:96]: Unauthorized it’s one of the response code in the flags

File: blogs/dns_tracking/dns_tracking.md (hunk 4 above)
Comment (msherif1234) [reply blogs/dns_tracking/dns_tracking.md:125]: No this for support tracking mDNS where need to deal with multiple responses

**Pull request closed without merging**